src/xpk/core/mtc.py

```diff
@@ -62,16 +68,111 @@ def install_mtc_on_cluster(args, system) -> int:
   if args.mtc_toleration_key is None:
     args.mtc_toleration_key = "google.com/tpu"
 
+  k8s_api_client = setup_k8s_env(args)
+  jobset_manifest = update_jobset_manifest()
+  if jobset_manifest is None:
+    xpk_print(
+        "No updating jobset controller as updated jobset manifest is empty."
+    )
+
+  xpk_print("Applying Jobset with MTC Configuration")
+  return_code = apply_kubectl_manifest(k8s_api_client, [jobset_manifest])
+  if return_code != 0:
+    return return_code
+
   mtc_checkpoint_configuration_crd_data = create_mtc_cpc(
       args.mtc_gcs_bucket,
       system.gce_machine_type,
       args.mtc_toleration_key,
       args.mtc_ramdisk_size,
   )
   xpk_print("Applying MTC Checkpoint Configuration")
-  k8s_api_client = setup_k8s_env(args)
   return_code = apply_kubectl_manifest(
       k8s_api_client, [mtc_checkpoint_configuration_crd_data]
   )
 
   return return_code
+
+
+def update_jobset_manifest():
+  manifest_url = f"https://github.com/kubernetes-sigs/jobset/releases/download/{JOBSET_VERSION}/manifests.yaml"
+  manifest_content = None
+  # Fetch the manifest content
+  try:
+    response = requests.get(manifest_url, timeout=10)
+    response.raise_for_status()  # Raise an exception for HTTP errors
+    manifest_content = response.text
+  except requests.exceptions.Timeout as e:
+    xpk_print(f"Error: Request to {manifest_url} after 10 seconds: {e}")
+    xpk_exit(1)
+  except requests.exceptions.RequestException as e:
+    xpk_print(f"Error fetching manifest from {manifest_url}: {e}")
+    xpk_exit(1)
+
+  if manifest_content is None:
+    xpk_print("Manifest content not found.")
+    xpk_exit(1)
+
+  # Load all YAML documents from the manifest
+  yaml_data_list = list(yaml.safe_load_all(manifest_content))
+  # Iterate through the yaml_data to find the Deployment for
+  # jobset-controller-manager
+  update_manifest = 0
+  for yaml_data in yaml_data_list:
+    if (
+        yaml_data
+        and yaml_data.get("apiVersion") == "apps/v1"
+        and yaml_data.get("kind") == "Deployment"
+        and yaml_data.get("metadata", {}).get("name")
+        == "jobset-controller-manager"
+    ):
+      # Found the Deployment, now modify the resources
+      containers = yaml_data["spec"]["template"]["spec"]["containers"]
+      for container in containers:
+        if container["name"] == "manager":
+          # Update resource limits and requests
+          current_cpu_request = (
+              container["resources"].get("requests", {}).get("cpu", "0m")
+          )
+          current_memory_request = (
+              container["resources"].get("requests", {}).get("memory", "0Mi")
+          )
+          current_memory_limit = (
+              container["resources"].get("limits", {}).get("memory", "0Mi")
+          )
+
+          # Define new values for comparison
+          new_cpu_request = "1000m"
+          new_memory_request = "1Gi"
+          new_memory_limit = "2Gi"
```

Comment:
Please add a readme comment on updated resource limits for the controller in an MTC cluster. https://github.com/AI-Hypercomputer/xpk/blob/develop/src/xpk/parser/cluster.py#L534 

If the user requests a really small CPU machine for the default pool, we want to make sure that jobset-controller can still be hosted.